File: 프로그래머스/lv2/12936. 줄 서는 방법/줄 서는 방법.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int n, long k) {
        int[] answer = new int[n];
        long temp = k - 1;
        ArrayList<Integer> list = new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            list.add(i);
        }

        for (int i = 0; i < n; i++) {
            if (temp == 0) {
                answer[i] = list.get(0);
                list.remove(0);
                continue;
            }
            long factor = factorial(n - i - 1);
            long num = temp / factor;
            answer[i] = list.get((int)num);
            list.remove((int)num);
            temp = temp % factor;
        }

        return answer;
    }

    public long factorial(int n) {
        long factor = 1;
        for (int i = 1; i <= n; i++) {
            factor *= i;
        }
        return factor;
    }
}

File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.py
def solution(n, k):
    answer = [0] * n
    n_list = []
    num = k - 1

    for i in range(1,n + 1):
        n_list.append(i)
        
    for j in range(0,n):
        if num == 0:
            answer[j] = n_list.pop(0)
            continue
        factor = factorial(n - j - 1)
        list_index = num // factor
        answer[j] = n_list.pop(list_index)
        num %= factor
        
    return answer

def factorial(k):
    factor = 1
    
    for i in range(1,k+1):
        factor *= i
        
    return factor